src/libraries/System.Data.Common/src/System/Data/DataSet.cs

```diff
@@ -227,6 +228,7 @@ protected DataSet(SerializationInfo info, StreamingContext context) : this(info,
         {
         }
 
+        [UnsupportedOSPlatform("browser")]
```

Comment:
This is same as #41085. It does not make sense to be marking serialization infrastructure as unsupported on browser.